storage_tiering_configuration.cpp

```diff
@@ -18,81 +18,65 @@ namespace irods {
-                                    plugin_spec_cfg.at("preserve_replicas"));
+                            preserve_replicas = plugin_spec_cfg.at("preserve_replicas");
                         }
 
                         if(plugin_spec_cfg.find("object_limit") != plugin_spec_cfg.end()) {
-                            object_limit = boost::any_cast<std::string>(
-                                    plugin_spec_cfg.at("object_limit"));
+                            object_limit = plugin_spec_cfg.at("object_limit");
                         }
 
                         if(plugin_spec_cfg.find("default_data_movement_parameters") != plugin_spec_cfg.end()) {
-                            default_data_movement_parameters = boost::any_cast<std::string>(
-                                    plugin_spec_cfg.at("default_data_movement_parameters"));
+                            default_data_movement_parameters = plugin_spec_cfg.at("default_data_movement_parameters");
                         }
 
                         if(plugin_spec_cfg.find("minimum_delay_time") != plugin_spec_cfg.end()) {
-                            default_data_movement_parameters = boost::any_cast<std::string>(
-                                    plugin_spec_cfg.at("minimum_delay_time"));
+                            default_data_movement_parameters = plugin_spec_cfg.at("minimum_delay_time");
                         }
 
                         if(plugin_spec_cfg.find("maximum_delay_time") != plugin_spec_cfg.end()) {
-                            default_data_movement_parameters = boost::any_cast<std::string>(
-                                    plugin_spec_cfg.at("maximum_delay_time"));
+                            default_data_movement_parameters = plugin_spec_cfg.at("maximum_delay_time");
                         }
 
                         if(plugin_spec_cfg.find("time_check_string") != plugin_spec_cfg.end()) {
-                            time_check_string = boost::any_cast<std::string>(
-                                    plugin_spec_cfg.at("time_check_string"));
+                            time_check_string = plugin_spec_cfg.at("time_check_string");
                         }
 
                         if(plugin_spec_cfg.find("number_of_scheduling_threads") != plugin_spec_cfg.end()) {
-                            number_of_scheduling_threads = boost::any_cast<int>(
-                                    plugin_spec_cfg.at("number_of_scheduling_threads"));
+                            number_of_scheduling_threads = plugin_spec_cfg.at("number_of_scheduling_threads");
                         }
 
                         if(plugin_spec_cfg.find(data_transfer_log_level_key) != plugin_spec_cfg.end()) {
-                            const std::string val = boost::any_cast<std::string>(
-                                    plugin_spec_cfg.at(data_transfer_log_level_key));
+                            const std::string val = plugin_spec_cfg.at(data_transfer_log_level_key);
```

Comment:
implicit casts need to be replaced.

use `get_ref<const std::string&>()` or `get<std::string>()` instead.

i'm guessing the previous lines need to be checked for implicit casts as well.